apm-lambda-extension/main.go

```diff
@@ -197,9 +141,9 @@ func main() {
 			defer timer.Stop()
```

Comment:
Implemented in 2870ee7. However, I had to move the Flush function out of that separate function to avoid a context deadlock after replacing `funcDone` by a context: 
- The invocation context could not be closed if the goroutine was still running due to the waitgroup
- The goroutine could not be interrupted by any other means than the invocation context being cancelled